patches/minecraft/net/minecraft/world/item/crafting/SimpleCookingSerializer.java.patch

```diff
@@ -18,4 +21,36 @@
 +      }
        float f = GsonHelper.m_13820_(p_44348_, "experience", 0.0F);
        int i = GsonHelper.m_13824_(p_44348_, "cookingtime", this.f_44327_);
-       return this.f_44328_.m_44352_(p_44347_, s, ingredient, itemstack, f, i);
+-      return this.f_44328_.m_44352_(p_44347_, s, ingredient, itemstack, f, i);
++      return this.f_44328_.create(p_44347_, s, ingredient, itemstack, f, i, new ResourceLocation(c));
+    }
+ 
+    public T m_8005_(ResourceLocation p_44350_, FriendlyByteBuf p_44351_) {
+       String s = p_44351_.m_130277_();
++      String c = p_44351_.m_130277_();
+       Ingredient ingredient = Ingredient.m_43940_(p_44351_);
+       ItemStack itemstack = p_44351_.m_130267_();
+       float f = p_44351_.readFloat();
+       int i = p_44351_.m_130242_();
+-      return this.f_44328_.m_44352_(p_44350_, s, ingredient, itemstack, f, i);
++      return this.f_44328_.create(p_44350_, s, ingredient, itemstack, f, i, new ResourceLocation(c));
+    }
+ 
+    public void m_6178_(FriendlyByteBuf p_44335_, T p_44336_) {
+       p_44335_.m_130070_(p_44336_.f_43728_);
++      if (p_44336_.category != null) {
++         p_44335_.m_130070_(p_44336_.category.toString());
++      } else {
++         p_44335_.m_130070_("");
++      }
+       p_44336_.f_43729_.m_43923_(p_44335_);
+       p_44335_.m_130055_(p_44336_.f_43730_);
+       p_44335_.writeFloat(p_44336_.f_43731_);
+@@ -49,6 +_,6 @@
+    }
+ 
+    interface CookieBaker<T extends AbstractCookingRecipe> {
+-      T m_44352_(ResourceLocation p_44353_, String p_44354_, Ingredient p_44355_, ItemStack p_44356_, float p_44357_, int p_44358_);
++      T create(ResourceLocation p_44353_, String p_44354_, Ingredient p_44355_, ItemStack p_44356_, float p_44357_, int p_44358_, ResourceLocation category);
```

Comment:
This also is a huge binary break. Any mod with a custom furnace variant is guaranteed to be broken by this.

You are probably best off making a second interface for the sake of the extra forge method

Reply:
Made a second interface so I believe I've fixed this.